Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -39,35 +108,25 @@ class MessageManager: EngineWebDelegate {
         }
     }
 
-    func showMessage(position: MessagePosition) {
-        elapsedTimer.start(title: "Displaying modal for message: \(currentMessage.messageId)")
-        messagePosition = position
+    // MARK: Timer determining how long message took to load.
+
+    // The manager subclasses are expected to call these functions to determine how long the messages took to load.
+
+    func startLoadingMessage() {
+        elapsedTimer.start(title: "Loading message with id: \(currentMessage.messageId)")
+    }
+
+    func doneLoadingMessage() {
+        elapsedTimer.end()
     }
 
     func getMessageView() -> GistView {
         isMessageEmbed = true
         return gistView
     }
 
-    private func loadModalMessage() {
-        if messageLoaded {
-            modalViewManager = ModalViewManager(gistView: gistView, position: messagePosition)
-            modalViewManager?.showModalView { [weak self] in
-                guard let self = self else { return }
-                self.delegate?.messageShown(message: self.currentMessage)
-                self.elapsedTimer.end()
-            }
-        }
-    }
-
     func dismissMessage(completionHandler: (() -> Void)? = nil) {
-        if let modalViewManager = modalViewManager {
-            modalViewManager.dismissModalView { [weak self] in
-                guard let self = self else { return }
-                self.delegate?.messageDismissed(message: self.currentMessage)
-                completionHandler?()
-            }
-        }
```

Comment:
This is code that should only be called for modal messages. So I moved this block into the Modal manager subclass.